### DISCUSSION TITLE
graphql POST occasional 502

### DISCUSSION THREAD

**Discussion by stiege** [requarks/wiki#4489] [Error / Bug Report] [answered]
I run wiki.js on a small home kubernetes cluster with a metallb load balancer and an nginx ingress.

```
"image": "requarks/wiki:latest"
```

I'm facing an issue in that while I'm editing a page, I'll quite often get a flash message:
![image](https://user-images.githubusercontent.com/6016056/134772568-da99d271-b92d-4c32-97d9-b1252e199461.png)

On inspecting chrome's logs I can see it's something to do with the `graphql` polling:
![image](https://user-images.githubusercontent.com/6016056/134772586-c77f9724-5555-47ea-bd03-9674fb3df6c9.png)
![image](https://user-images.githubusercontent.com/6016056/134772597-30da3164-8379-444a-8049-e8010cb375c9.png)

I have tried "Copy as cURL (bash)" but have been unable to reproduce this outside my web browser.

The usual response coming back is simply:
```
[{"data":{"pages":{"checkConflicts":false,"__typename":"PageQuery"}}}]
```
With headers:
```
HTTP/1.1 200 OK
Date: Sat, 25 Sep 2021 20:55:20 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 71
Connection: keep-alive
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
X-UA-Compatible: IE=edge
Referrer-Policy: same-origin
ETag: W/"47-eAEMDnCWeLcnynCdoLq6pAlYt5w"
Vary: Accept-Encoding
```

While when it fails it gets the content:
```
<html>
<head><title>502 Bad Gateway</title></head>
<body>
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx</center>
</body>
</html>
<!-- a padding to disable MSIE and Chrome friendly error page -->
<!-- a padding to disable MSIE and Chrome friendly error page -->
<!-- a padding to disable MSIE and Chrome friendly error page -->
<!-- a padding to disable MSIE and Chrome friendly error page -->
<!-- a padding to disable MSIE and Chrome friendly error page -->
<!-- a padding to disable MSIE and Chrome friendly error page -->
```

With headers:
```
HTTP/1.1 502 Bad Gateway
Date: Sat, 25 Sep 2021 20:55:15 GMT
Content-Type: text/html
Content-Length: 552
Connection: keep-alive
```

From nginx-ingress logs I do see:

```
2021/09/25 21:27:00 [error] 47#47: *150132 upstream prematurely closed connection while reading response header from upstream, client: 192.168.178.37, server: wiki.london.vpn, request: "POST /graphql HTTP/1.1", upstream: "http://172.16.96.139:3000/graphql", host: "wiki.london.vpn", referrer: "http://wiki.london.vpn/e/en/home"
```

I was also unable to repro the error by making the same request manually from the `/graphql` endpoint in the web browser.

![image](https://user-images.githubusercontent.com/6016056/134776444-efdada6e-c4d5-4ea2-967c-9f3affb6fb22.png)

Any help appreciated.

**Comment by RachadAbiChahine** [comment]
I gone back to older version to get back the git content into the postgreSQL then I upgrade and now its working but sure its a regression :)

**Comment by stiege** [comment, marked as answer]
After setting my imagepullpolicy to "Always" I think a newer container has resolved the issue - unable to reproduce. I will point to `2.5` rather than `latest`.
